homeassistant/components/wake_on_lan.py

```diff
@@ -0,0 +1,60 @@
+"""
+Component to wake up devices sending Wake-On-LAN magic packets.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/wake_on_lan/
+"""
+import asyncio
+import logging
+import os
+
+import voluptuous as vol
+
+from homeassistant.config import load_yaml_config_file
+from homeassistant.const import CONF_MAC
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['wakeonlan==0.2.2']
+
+DOMAIN = "wake_on_lan"
+_LOGGER = logging.getLogger(__name__)
+
+CONF_BROADCAST_ADDRESS = 'broadcast_address'
+
+SERVICE_SEND_MAGIC_PACKET = 'send_magic_packet'
+
+WAKE_ON_LAN_SEND_MAGIC_PACKET_SCHEMA = vol.Schema({
+    vol.Required(CONF_MAC): cv.string,
+    vol.Optional(CONF_BROADCAST_ADDRESS): cv.string,
+})
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Set up the wake on LAN component."""
+    from wakeonlan import wol
+
+    @callback
```

Comment:
Ok, but, should I make it a coroutine and `yield from hass.async_add_job(wol.send_magic_packet`? What's the preferred method?